### ISSUE TITLE
About multiline lists and maps

### ISSUE THREAD

**Issue by valeriangalliat** [HugoGiraudel/sass-guidelines#56]
Current guidelines, about lists:

> do not add a trailing comma. It is ugly and it does not help.

And maps:

> comma (,) at the end of each key/value except the last;

I agree with these guidelines when the list/map fits in a signle line, but I like adding a trailing comma for multiline lists and maps. It allows to keep clean diffs when moving, appending or deleting lines, and don't require the developer (or a script) to handle differently the last line when editing code (this is annoying when using `sed`, multicursors, range substitution or macros for example).

I know this is not something everybody likes, but at least the two style guides I follow for [JavaScript](http://nodeguide.com/style.html#object-array-creation) and [PHP](http://symfony.com/doc/current/contributing/code/standards.html#structure) are recommending this too, and I see Rust is looking forward to [recommend this practice](https://github.com/rust-lang/rust-guidelines/blob/09d1b8fc409b2707166c7688f41da7554d72bf41/style/braces.md#trailing-commas) too.

So basically, I'm for the following (let's assume the example strings are long enough to require a multiline structure):

```scss
$foo: (
  "a": "b",
  "c": "d",
);

$bar: (
  "a",
  "b",
);
```

However, I would be okay with not having commas at all in a multiline list.

Opinions?

**Comment by HugoGiraudel** [comment]
Multi-lines lists are prohibited anyway. Although I am thinking of forcing a trailing comma on maps.

**Comment by valeriangalliat** [comment]
> Multi-lines lists are prohibited anyway.

I don't recall reading this, maybe it should be added?

**Comment by tobysteward** [comment]
Multiline's are coming - see https://github.com/sass/sass/issues/216

**Comment by valeriangalliat** [comment]
Oh it's prohibited at Sass level. I misunderstood Hugo's comment.

Well if it's comming, we can still anticipate this!

**Comment by HugoGiraudel** [comment]
> I don't recall reading this, maybe it should be added?

I should make this clear. I don't think it is a good idea to use multiple lines for a list unless it has too many items to fit on a 80-characters long line.

I'll add this.

**Comment by HugoGiraudel** [comment]
You got it.
